src/components/shared/chart/Viewport.js

```diff
@@ -228,10 +228,12 @@ export const withChartViewport: WithChartViewport<*, *> =
       _lastKeyboardNavigationFrame: number = 0;
       _keysDown: Set<NavigationKey> = new Set();
       _deltaToZoomFactor = delta => Math.pow(ZOOM_SPEED, delta);
+      _resizeObserver: ResizeObserver;
 
       constructor(props: ViewportProps) {
         super(props);
         this.state = this.getDefaultState(props);
+        this._resizeObserver = new ResizeObserver(this._setSize);
```

Comment:
Question: I'm curious, why prefer this? I'm unfamiliar with ResizeObserver.

Reply:
I need to do some more research, and try some use cases, but I believe the main advantages are:
* Callback is called after layout but before paint (I believe resize event handlers are called before layout, so we need to use rAF, but that's not generally necessary with ResizeObserver). 
* I think it's called also when the observed element is resized for other reasons than window resizes (when it's resized because of a relayout)

I still have to double check cases like the full page zoom.

Hope that helps :-)